processor.go

```diff
@@ -14,8 +15,9 @@ import (
 // A Processor handle the service processing and execute hooks on the resulting
 // fields.
 type Processor struct {
-	lock  sync.Mutex
-	hooks []Hook
+	lock    sync.Mutex
+	hooks   []Hook
+	hooksEx []HookEx
```

Comment:
Indeed, that would work, but it may break existing code that would already implement an `Init(ctx context.Context)` method and doesn't expect this method to be called. That may be acceptable or not and also depends if we want to have a major or minor version bump.